pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -417,16 +408,15 @@ def on_validation_end(self, trainer, pl_module):
             )
             filepath = os.path.join(self.dirpath, f"{filename}.ckpt")
             self._save_model(filepath, trainer, pl_module)
-            if self.last_model_path and self.last_model_path != filepath:
+            if (
+                self.last_model_path
+                and self.last_model_path != filepath
+                and trainer.is_global_zero
+            ):
                 self._del_model(self.last_model_path)
 
     def _do_check_save(
-        self,
-        filepath: str,
-        current: torch.Tensor,
-        epoch: int,
-        trainer,
-        pl_module,
+        self, filepath: str, current: torch.Tensor, epoch: int, trainer, pl_module
```

Comment:
the Black inconsistency is funny...

Reply:
this might've been my bad rebasing before